### ISSUE TITLE
Redirect loop

### ISSUE THREAD

**Issue by teneightfive** [UKHomeOffice/hof#46]
If you curl one of the sub app routes it currently ends in a redirect loop. 

```
curl -v -L https://www.biometric-residence-permit.service.gov.uk/collection/ > /dev/null
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 54.77.114.221...
* Connected to www.biometric-residence-permit.service.gov.uk (54.77.114.221) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
* Server certificate: www.biometric-residence-permit.service.gov.uk
* Server certificate: QuoVadis Global SSL ICA G2
* Server certificate: QuoVadis Root CA 2
> GET /collection/ HTTP/1.1
> Host: www.biometric-residence-permit.service.gov.uk
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 302 Moved Temporarily
< Server: openresty
< Date: Fri, 26 Feb 2016 16:57:19 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 51
< Connection: keep-alive
< X-Powered-By: Express
< set-cookie: hmpo-wizard-sc=1; Path=/; HttpOnly; Secure
< set-cookie: hmbrp.sid=s%3AvEcQDKHcdV-2g-ZwLtGVxJ0lip8EH6c-.68LKXo5gDoF71SG57K4ZxjjHVKBB3hz9k3IoXNyTcKg; Path=/; HttpOnly; Secure
< Location: /collection/where
< Vary: Accept
< Cache-Control: private
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
<
* Ignoring the response-body
{ [51 bytes data]
100    51  100    51    0     0    286      0 --:--:-- --:--:-- --:--:--   288
* Connection #0 to host www.biometric-residence-permit.service.gov.uk left intact
* Issue another request to this URL: 'https://www.biometric-residence-permit.service.gov.uk/collection/where'
* Found bundle for host www.biometric-residence-permit.service.gov.uk: 0x7f9f7a410110
* Re-using existing connection! (#0) with host www.biometric-residence-permit.service.gov.uk
* Connected to www.biometric-residence-permit.service.gov.uk (54.77.114.221) port 443 (#0)
> GET /collection/where HTTP/1.1
> Host: www.biometric-residence-permit.service.gov.uk
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 302 Moved Temporarily
< Server: openresty
< Date: Fri, 26 Feb 2016 16:57:19 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 46
< Connection: keep-alive
< X-Powered-By: Express
< set-cookie: hmpo-wizard-sc=1; Path=/; HttpOnly; Secure
< set-cookie: hmbrp.sid=s%3ASV6gQ7EjlAdjmilyIFXyEix18Djn-Qs4.68LKXo5gDoF71SG57K4ZxjjHVKBB3hz9k3IoXNyTcKg; Path=/; HttpOnly; Secure
< Location: /collection/
< Vary: Accept
< Cache-Control: private
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
<
* Ignoring the response-body
{ [46 bytes data]
100    46  100    46    0     0    216      0 --:--:-- --:--:-- --:--:--   216
* Connection #0 to host www.biometric-residence-permit.service.gov.uk left intact
* Issue another request to this URL: 'https://www.biometric-residence-permit.service.gov.uk/collection/'
* Found bundle for host www.biometric-residence-permit.service.gov.uk: 0x7f9f7a410110
* Re-using existing connection! (#0) with host www.biometric-residence-permit.service.gov.uk
* Connected to www.biometric-residence-permit.service.gov.uk (54.77.114.221) port 443 (#0)
> GET /collection/ HTTP/1.1
> Host: www.biometric-residence-permit.service.gov.uk
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 302 Moved Temporarily
< Server: openresty
< Date: Fri, 26 Feb 2016 16:57:19 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 51
< Connection: keep-alive
< X-Powered-By: Express
< set-cookie: hmpo-wizard-sc=1; Path=/; HttpOnly; Secure
< set-cookie: hmbrp.sid=s%3A4FIqqJtg2MKn5zFsY8CIYqQcy71uf3H1.SuPcanMsk6gcDMlRmFfla1gGv%2BsL8aHDiskt6itza7s; Path=/; HttpOnly; Secure
< Location: /collection/where
< Vary: Accept
< Cache-Control: private
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
<
* Ignoring the response-body
{ [51 bytes data]
100    51  100    51    0     0    208      0 --:--:-- --:--:-- --:--:--   208
* Connection #0 to host www.biometric-residence-permit.service.gov.uk left intact
* Issue another request to this URL: 'https://www.biometric-residence-permit.service.gov.uk/collection/where'
* Found bundle for host www.biometric-residence-permit.service.gov.uk: 0x7f9f7a410110
* Re-using existing connection! (#0) with host www.biometric-residence-permit.service.gov.uk
* Connected to www.biometric-residence-permit.service.gov.uk (54.77.114.221) port 443 (#0)
> GET /collection/where HTTP/1.1
> Host: www.biometric-residence-permit.service.gov.uk
> User-Agent: curl/7.43.0
> Accept: */*
>
< HTTP/1.1 302 Moved Temporarily
< Server: openresty
< Date: Fri, 26 Feb 2016 16:57:19 GMT
< Content-Type: text/plain; charset=utf-8
< Content-Length: 46
< Connection: keep-alive
< X-Powered-By: Express
< set-cookie: hmpo-wizard-sc=1; Path=/; HttpOnly; Secure
< set-cookie: hmbrp.sid=s%3AEXHkuUIR7olry5JL__3AJG65jF7Ck7LA.68LKXo5gDoF71SG57K4ZxjjHVKBB3hz9k3IoXNyTcKg; Path=/; HttpOnly; Secure
< Location: /collection/
< Vary: Accept
< Cache-Control: private
< X-Frame-Options: SAMEORIGIN
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
<
* Ignoring the response-body
{ [46 bytes data]
```

this continues a few more times, have only pasted a snippet of the output.

**Comment by timgent** [comment]
We've recently found this too and Ben has just raised a PR for BRP:
https://github.com/UKHomeOffice/brp_app/pull/348

The issue is to do with redirecting when cookies can't be set, which can end in an infinite loop.

Also just to note the issue is with how the example app has been put together, rather than the hof libraries themselves. We've got a few changes we'd like to make to improve the example app, just not had a chance to update it yet.
